pkgs/applications/networking/cluster/calico/cni.nix

```diff
@@ -0,0 +1,36 @@
+{ stdenv, buildGoPackage, fetchFromGitHub }:
+
+buildGoPackage rec {
+  name = "calico-cni-${version}";
+  version = "1.4.0-rc3";
+
+  src = fetchFromGitHub {
+    owner = "projectcalico";
+    repo = "calico-cni";
+    rev = "v${version}";
+    sha256 = "1a67zfqgl2ngpbfqzafvwrd0s5gwyba39ifr5j5vp93xkv1ckn98";
+  };
+
+  outputs = ["out" "bin" "plugins"];
+
+  goPackagePath = "github.com/calico-cni";
+  goDeps = ./deps.json;
+
+  # Remove vendor folders
+  preBuild = ''find go/src -type d -name 'vendor' -prune -exec rm -r {} \;'';
```

Comment:
I don't, only of dependencies. There were build issues with vendored packages of dependencies, because of different versions in vendor packages in root and in dependencies.